plugins/internal/telemetry/http_transport.go

```diff
@@ -0,0 +1,121 @@
+package telemetry
+
+import (
+	"io"
+	"net/http"
+	"sync/atomic"
+	"time"
+
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/metric"
+)
+
+// Refer OpenTelemetry Semantic Conventions for HTTP Client.
+// https://github.com/open-telemetry/semantic-conventions/blob/main/docs/http/http-metrics.md#http-client
+const (
+	metricClientDuration     = "http.client.duration"
+	metricClientRequestSize  = "http.client.request.size"
+	metricClientResponseSize = "http.client.response.size"
+
+	attributeServerPort         = "server.port"
+	attributeServerAddress      = "server.address"
+	attributeRequestMethod      = "http.request.method"
+	attributeResponseStatusCode = "http.response.status_code"
+)
+
+func NewHTTPTransport(baseTransport http.RoundTripper) http.RoundTripper {
+	if _, ok := baseTransport.(*httpTransport); ok {
+		return baseTransport
+	}
+
+	if baseTransport == nil {
+		baseTransport = http.DefaultTransport
+	}
+
+	icl := &httpTransport{roundTripper: baseTransport}
+	icl.createMeasures(otel.Meter("github.com/goto/meteor/plugins/internal/telemetry"))
+
+	return icl
+}
+
+type httpTransport struct {
+	roundTripper http.RoundTripper
+
+	counters       map[string]metric.Int64Counter
+	valueRecorders map[string]metric.Float64Histogram
```

Comment:
Not a hard change request but it might be simpler and clearer to just have the actual counters and histogram instead of maps. We would only have 1 more field in the struct.

Reply:
hmm makes sense. i was mostly following the metrics PR being added for net/http .